test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExpectToExceedThresholdWeightControllerSpec.scala

```diff
@@ -0,0 +1,214 @@
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.json.JsObject
+import play.api.test.Helpers.{redirectLocation, status}
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.plasticpackagingtax.registration.connectors.DownstreamServiceError
+import uk.gov.hmrc.plasticpackagingtax.registration.controllers.{routes => pptRoutes}
+import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.LiabilityDetails
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.liability.expect_to_exceed_threshold_weight_page
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+
+class ExpectToExceedThresholdWeightControllerSpec extends ControllerSpec {
+  private val page = mock[expect_to_exceed_threshold_weight_page]
+  private val mcc  = stubMessagesControllerComponents()
+
+  private val controller =
+    new ExpectToExceedThresholdWeightController(authenticate = mockAuthAction,
+                                                mockJourneyAction,
+                                                mockRegistrationConnector,
+                                                mcc = mcc,
+                                                page = page
+    )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    when(page.apply(any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(page)
+    super.afterEach()
+  }
+
+  "Liability Expect To Exceed Threshold Weight Controller" should {
+
+    "return 200" when {
+
+      "user is authorised and display page method is invoked" in {
+        authorizedUser()
+        mockRegistrationFind(aRegistration())
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+
+      "user is authorised, a registration already exists and display page method is invoked" in {
+        val registration =
+          aRegistration(
+            withLiabilityDetails(LiabilityDetails(expectToExceedThresholdWeight = Some(true)))
+          )
+        authorizedUser()
+        mockRegistrationFind(registration)
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+
+      "user is authorised, a registration already exists and display page method without the expectToExceedThresholdWeight not set" in {
+        val registration =
+          aRegistration(
+            withLiabilityDetails(LiabilityDetails(expectToExceedThresholdWeight = None))
+          )
+        authorizedUser()
+        mockRegistrationFind(registration)
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+    }
+
+    forAll(Seq(saveAndContinueFormAction, saveAndComeBackLaterFormAction)) { formAction =>
+      "return 303 (OK) for " + formAction._1 when {
+        "user submits 'Yes' answer with isPreLaunch enabled" in {
```

Comment:
As this controller operates on a post-launch only URL now, I think we can ditch the feature switch specific tests?